File: leetcode/Easy/1.two-sum.cpp
#include <algorithm>
#include <iostream>
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
   public:
    vector<int> twoSum(vector<int> &nums, int target) {
        unordered_map<int, int> hash;
        for (int i = 0, n = nums.size(); i < n; ++i) {
            if (hash.find(target - nums[i]) != hash.end())
                return {i, hash[target - nums[i]]};
            hash[nums[i]] = i;
        }
        return {-1, -1};
    }
};

int main() {
    Solution sol;

    return 0;
}

File: leetcode/Easy/1.two-sum.go
package main
import "fmt"
func twoSum(nums []int, target int) []int {
  var hash = make(map[int]int)
  for i, v := range nums {
    _, exists := hash[target - v]
    if (exists) {
      return []int{hash[target - v], i}
    }
    hash[v] = i
  }
  return []int{-1, -1};
}

func main(){
    nums := []int{2,7,11,15}
    fmt.Println(twoSum(nums, 9));
}
